project-minimization-plugin/src/main/kotlin/org/plan/research/minimization/plugin/psi/PsiUtils.kt

```diff
@@ -63,6 +67,23 @@ object PsiUtils {
         return PsiChildrenIndexDDItem.create(element, parentPath, localPath, renderedType)
     }
 
+    /**
+     * Transforms PsiElement into **deletable** PsiDDItem by traversing the parents and collecting file information
+     *
+     * @param element Element to transform
+     * @param context
+     * @return a converted PSIDDItem
+     */
+    @RequiresReadLock
+    fun buildDeletablePsiItem(
+        context: IJDDContext,
+        element: PsiElement,
+    ): Option<PsiStubDDItem> = option {
+        val (currentFile, parentPath) = buildParentPath(element, { _, element -> KtStub.create(element) }) { true }!!
```

Comment:
That's a very complicated question. While developing, I was thinking that there was some natural restriction. For example, since all variables in code have type `KtProperty` we need to somehow restrict the variation (for example, all values in class initializers are not valid). So, stubs somehow restrict it. 

On the other side, this restriction might be a stopping point: for example, we will not try to delete the function in the body of lambda. Additionally, it's a complicated code that could be hard to understand.

So, I personally chosen the restriction variant. WDYT for those reasons?